Mathlib/SetTheory/Game/Birthday.lean

```diff
@@ -157,26 +161,121 @@ theorem birthday_add : ∀ x y : PGame.{u}, (x + y).birthday = x.birthday ♯ y.
     · exact lt_max_of_lt_right ((nadd_le_nadd_left hj _).trans_lt (lt_lsub _ _))
 termination_by a b => (a, b)
 
-theorem birthday_add_zero : (a + 0).birthday = a.birthday := by simp
-
-theorem birthday_zero_add : (0 + a).birthday = a.birthday := by simp
-
-theorem birthday_add_one : (a + 1).birthday = Order.succ a.birthday := by simp
-
-theorem birthday_one_add : (1 + a).birthday = Order.succ a.birthday := by simp
+@[simp]
+theorem birthday_sub (x y : PGame) : (x - y).birthday = x.birthday ♯ y.birthday := by
+  apply (birthday_add x _).trans
+  rw [birthday_neg]
 
 @[simp]
 theorem birthday_natCast : ∀ n : ℕ, birthday n = n
   | 0 => birthday_zero
   | n + 1 => by simp [birthday_natCast]
 
-@[deprecated (since := "2024-04-17")]
-alias birthday_nat_cast := birthday_natCast
+end PGame
+
+namespace Game
 
-theorem birthday_add_nat (n : ℕ) : (a + n).birthday = a.birthday + n := by simp
+/-- The birthday of a game is defined as the least birthday among all pre-games that define it. -/
+noncomputable def birthday (x : Game.{u}) : Ordinal.{u} :=
+  sInf (PGame.birthday '' (Quotient.mk' ⁻¹' {x}))
 
-theorem birthday_nat_add (n : ℕ) : (↑n + a).birthday = a.birthday + n := by simp
+/-- The set in the definition of birthday is nonempty. -/
+theorem birthday_nonempty (x : Game) : (PGame.birthday '' (Quotient.mk' ⁻¹' {x})).Nonempty := by
```

Comment:
Should this be a theorem? This is trivial, and you only use it once

Reply:
The intention of this theorem is that if someone wants to work with `birthday` using the direct infimum definition, they can show that they're not in the trivial and somewhat badly behaved case of `sInf Ø`. Admittedly I'm not sure how often you'll want to do this, but even if it's never, I don't think having this theorem hurts.

Reply:
In those situations, what we usually do is make the lemma `private`. The rationale is that people shouldn't this kind of lemma past the API, else there is a problem with the API.

Reply:
I don't remember if I was the one who added them, but there do exist other lemmas like this for the ordinal API, like [`Ordinal.sub_nonempty`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/SetTheory/Ordinal/Arithmetic.html#Ordinal.sub_nonempty) or [`Ordinal.div_nonempty`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/SetTheory/Ordinal/Arithmetic.html#Ordinal.div_nonempty). Should those be privated in a separate PR?

Reply:
Yeah probably